packages/devextreme/js/__internal/ui/button/props.ts

```diff
@@ -0,0 +1,50 @@
+import type { BaseWidgetProps } from '@ts/core/r1/base_props';
+import { BaseWidgetDefaultProps } from '@ts/core/r1/base_props';
+
+export interface ButtonProps extends BaseWidgetProps {
+  activeStateEnabled: boolean;
+
+  hoverStateEnabled: boolean;
+
+  icon: string;
+
+  iconPosition?: string;
+
+  onClick?: (e: { event: Event }) => void;
+
+  onSubmit?: (e: { event: Event; submitInput: HTMLInputElement | null }) => void;
+
+  pressed?: boolean;
+
+  stylingMode: 'outlined' | 'text' | 'contained';
+
+  template?: (props: { data: { icon?: string; text?: string } }) => JSX.Element;
+
+  iconTemplate?: (props) => JSX.Element;
+
+  children?: JSX.Element;
+
+  text: string;
+
+  type: 'danger' | 'default' | 'normal' | 'success';
```

Comment:
Shouldn't there be types from common.d.ts?

Reply:
I moved as it was in button.tsx: https://github.com/DevExpress/DevExtreme/blob/24_2/packages/devextreme/js/renovation/ui/button.tsx#L137